src/httpClient/httpClient.ts

```diff
@@ -243,7 +226,7 @@ export default class HttpClient {
     }
 
     return {
-      ...headers,
+      ...(headers || {}),
```

Comment:
I see `headers` are defaulted to `{}` already in the param definition. Do we need this fallback + the null check on ln 218?